### PR TITLE
Add genesis federate check for free bridge transactions.

### PR DESCRIPTION
**Pull request by tinchou**

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java [hunk 1]
```diff
@@ -158,14 +159,15 @@ public static boolean isFreeBridgeTx(org.ethereum.core.Transaction rskTx, long b
...
-                       isFromFederateMember(rskTx) ||
+                       isFromGenesisFederateMember(rskTx, bridgeConstants) ||
                        isFromFederationChangeAuthorizedSender(rskTx, bridgeConstants) ||
                        isFromLockWhitelistChangeAuthorizedSender(rskTx, bridgeConstants)
                );
     }
 
-    private static boolean isFromFederateMember(org.ethereum.core.Transaction rskTx) {
-        return true;
+    private static boolean isFromGenesisFederateMember(org.ethereum.core.Transaction rskTx, BridgeConstants bridgeConfiguration) {
```
Comment (amendelzon) [inline rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java]: This sounds kind of odd to me. The method is called `isFromGenesisFederateMember` but needs to be fed the genesis federation from outside. So either one of two things:

1 - It doesn't actually have knowledge about the genesis federation, therefore the name is wrong. Remove the `Genesis` bit and pass the federation that is being checked for membership.
2 - Let the method gather the genesis federation itself.

I personally vote on option 1.

Cheers.

**Comment by rskops** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 2 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeUtils.java#L42](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L42): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L95](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L95): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L124](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L124): Remove this unused method parameter "bridgeConstants". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L131](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L131): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L158](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L158): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [Federation.java#L149](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/Federation.java#L149): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [Federation.java#L151](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/Federation.java#L151): Method co.rsk.peg.Federation.equals(Object) orders expressions in a conditional in a sub optimal way [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ASEO_SUBOPTIMAL_EXPRESSION_ORDER)
1. ![MINOR][MINOR] [Federation.java#L132](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/Federation.java#L132): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![MINOR][MINOR] [Federation.java#L135](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/Federation.java#L135): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![INFO][INFO] [BridgeUtils.java#L96](https://github.com/rsksmart/rskj/blob/8f8ed7c7e0e3be0191b6909eb425bcc1588bd350/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L96): method co.rsk.peg.BridgeUtils.scriptCorrectlySpendsTx(BtcTransaction, int, Script) converts an exception into a boolean 'error code' value [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AEXS_EXCEPTION_SOFTENING_RETURN_FALSE)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java (hunk 1 above)
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java]: I think I was following the other methods conventions, but I like your suggestion better. Just pushed and amended commit!

Comment (amendelzon) [reply rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java]: Yeah, was just looking and both `isFromFederationChangeAuthorizedSender` and `isFromLockWhitelistChangeAuthorizedSender` probably need the same treatment. Future refactoring.

**Comment by rskops** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 2 minor
* ![INFO][INFO] 2 info

#### Top 10 issues

1. ![MAJOR][MAJOR] [BridgeUtils.java#L42](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L42): Add a private constructor to hide the implicit public one. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1118)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L95](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L95): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L124](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L124): Remove this unused method parameter "bridgeConstants". [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L131](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L131): Either log or rethrow this exception. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BridgeUtils.java#L158](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L158): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [Federation.java#L149](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/Federation.java#L149): Reduce the number of conditional operators (5) used in the expression (maximum allowed 3). [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS1067)
1. ![MAJOR][MAJOR] [Federation.java#L151](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/Federation.java#L151): Method co.rsk.peg.Federation.equals(Object) orders expressions in a conditional in a sub optimal way [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3ASEO_SUBOPTIMAL_EXPRESSION_ORDER)
1. ![MINOR][MINOR] [Federation.java#L132](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/Federation.java#L132): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![MINOR][MINOR] [Federation.java#L135](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/Federation.java#L135): Missing curly brace. [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=squid%3AS00121)
1. ![INFO][INFO] [BridgeUtils.java#L96](https://github.com/rsksmart/rskj/blob/84347192fed442c28a62b9ca98e98a3591233626/rskj-core/src/main/java/co/rsk/peg/BridgeUtils.java#L96): method co.rsk.peg.BridgeUtils.scriptCorrectlySpendsTx(BtcTransaction, int, Script) converts an exception into a boolean 'error code' value [rule](https://jenkins.rsklabs.io/sonar/coding_rules#rule_key=fb-contrib%3AEXS_EXCEPTION_SOFTENING_RETURN_FALSE)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**